docker/docker_client.go

```diff
@@ -78,19 +88,21 @@ func newDockerClient(refHostname, certPath string, tlsVerify bool) (*dockerClien
 	}, nil
 }
 
-// makeRequest creates and executes a http.Request with the specified parameters, adding authentication and TLS options for the Docker client.
-// url is NOT an absolute URL, but a path relative to the /v2/ top-level API path.  The host name and schema is taken from the client or autodetected.
-func (c *dockerClient) makeRequest(method, url string, headers map[string][]string, stream io.Reader) (*http.Response, error) {
-	if c.scheme == "" {
-		pr, err := c.ping()
-		if err != nil {
-			return nil, err
+// MakeRequest creates and executes a http.Request with the specified parameters, adding authentication and TLS options for the Docker client.
+// if resolved is fals, eurl is NOT an absolute URL, but a path relative to the /v2/ top-level API path.  The host name and schema is taken from the client or autodetected.
+func (c *dockerClient) MakeRequest(method, url string, headers map[string][]string, stream io.Reader, resolved bool) (*http.Response, error) {
```

Comment:
Is your _intent_ to make the `makeRequest` logic overridable? And fully overridable, or just the ability to make it go away so that we don’t have to mock a response to `c.ping`?

Or do you really want to abstract only the underlying `makeRequestToResolvedURL`, i.e. to keep `dockerClient` but then have a `dockerClientRequestMaker`? That one would start looking awfully like a `http.RoundTripper`, see how k8s uses `http.RoundTripper` in https://github.com/kubernetes/kubernetes/blob/master/pkg/client/transport/round_trippers.go#L32 to do basically what our `makeRequestToResolvedURL` does.